hardware_interface/include/hardware_interface/hardware_info.hpp

```diff
@@ -48,16 +52,16 @@ struct ComponentInfo
 {
   /// Name of the component.
   std::string name;
-  /// Type of the component: sensor or joint.
+  /// Type of the component: sensor, joint, or GPIO.
   std::string type;
   /**
    * Name of the command interfaces that can be set, e.g. "position", "velocity", etc.
-   * Used by joints.
+   * Used by joints and GPIOs.
    */
   std::vector<InterfaceInfo> command_interfaces;
   /**
    * Name of the state interfaces that can be read, e.g. "position", "velocity", etc.
-   * Used by Joints and Sensors.
+   * Used by Joints, Sensors and GPIOs.
```

Comment:
Nit-pick: let's keep the style...

```suggestion
   * Used by joints, sensors and GPIOs.
```